### DISCUSSION TITLE
how  to install peer depenencies ?

### DISCUSSION THREAD

**Discussion by vaynevayne** [storybookjs/storybook#23746] [Help] [answered]
i use monorepo make ui-library , it use react-table as peerDepenencies, in storybook app, how to install it to demo?

**Comment by future-leader1** [comment]
Sure thing! It sounds like you're working on a monorepo with a UI library that uses react-table as a peer dependency. And you want to set up a Storybook app to showcase your UI library. Let's break down the steps to get your demo up and running:

Install Storybook: If you haven't already, you need to install Storybook in your monorepo. You can do this by running:
bash
Copy code
npx sb init
This will guide you through the setup process and create the necessary files and configuration for Storybook.
Add Dependency: Since your UI library uses react-table as a peer dependency, you need to make sure it's installed in your Storybook app as well. You can do this by running:
bash
Copy code
npm install react-table
Configure Webpack: Sometimes, peer dependencies need a bit of extra configuration to work with Storybook. In your Storybook's .storybook directory, you might find a webpack.config.js file. Make sure you're extending your main monorepo's Webpack configuration so that the peer dependencies are resolved correctly.
Component Imports: Now, you can start creating your Storybook stories to showcase your UI library components. Import your UI components and use them to create stories. For example:
jsx
Copy code
import React from 'react';
import { storiesOf } from '@storybook/react';
import { YourComponent } from 'path/to/your/ui-library';

storiesOf('Your Component', module)
  .add('Default', () => <YourComponent />);
Run Storybook: To see your components in action, run the Storybook app:
bash
Copy code
npm run storybook
This should start a local development server, and you can access your Storybook at http://localhost:6006.
View Your Demo: With the Storybook server running, you should be able to see and interact with your UI library components in a nice showcase environment. You can navigate through the different stories you've created and see how your components behave.
Remember that these are general steps, and your actual setup might have some variations depending on your monorepo's structure, configuration, and other specific factors. If you encounter any issues, don't hesitate to ask for help or provide more details for more specific guidance. Good luck with your UI library demo!

**Comment by vaynevayne** [comment]
thanks, resolved,  I need to install the dependencies in @org/input instead of @org/storybook
```jsx
cd components/input
pnpm add @transtack/react-table --save-peer
```

When I run sb dev, it automatically installs peerdependence, right? @future-leader1

**Comment by future-leader1** [comment, marked as answer]
yes probably , When you run sb dev or npm run storybook, Storybook usually handles dependencies and installs any needed peer dependencies for your components. If you've set up @transtack/react-table correctly as a peer dependency for @org/input, and your Storybook config is right, sb dev should handle the dependencies fine.
